ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -521,43 +521,44 @@ public NamedTopology getNamedTopology(final SharedKafkaStreamsRuntime sharedRunt
   }
 
   private SharedKafkaStreamsRuntime getKafkaStreamsInstance(
+          final String applicationId,
           final Set<SourceName> sources,
-          final QueryId queryID,
+          final QueryId queryId,
           final MetricCollectors metricCollectors) {
     for (final SharedKafkaStreamsRuntime sharedKafkaStreamsRuntime : streams) {
```

Comment:
Why do we need to track the `streams` again in the `QueryBuilder` class if we only ever need to access a single SharedKafkaStreamsRuntime per query (and there's one QueryBuilder per query, IIUC)? It seems like we could remove that field entirely, as well as the `real` flag, and then just look up the correct runtime inside QueryRegistryImpl and pass it in as a parameter to the QueryBuilder

(Also, instead of looping through all of the streams can't we use a map keyed by the application/runtime id?)